bowling-ball/Game.cs

```diff
@@ -1,17 +1,76 @@
-﻿using System;
+using System;
 
 namespace BowlingBall
 {
     public class Game
     {
-        public void Roll(int pins)
-        {
-            throw new NotImplementedException();
+        //to store points in the game
+        int[] frame = new int[21];
+        //for accessing frame array for storing the rolled pins
+        int framePointer = 0;
+        //for maintaining score
+        int totalScore = 0;
+        public void Roll(int pins)
+        {
+            if (pins < 0 || pins > 10)
+                throw new ArgumentException("Enter pin between 1 to 10 both inclusive");
+
+            frame[framePointer++] = pins;
+        }
+
+        public int GetScore()
+        {
+            int index = 0;
+            for(int frameNumber = 0; frameNumber < 10; frameNumber++)
+            {
+                // Case when it is a strike
+                if(IsStrike(index))
+                {
+                    totalScore = totalScore + 10 + BonusForStrike(index);
+                    index++;
+                }
+                // Case when it is a spare
+                else if(IsSpare(index))
+                {
+                    totalScore = totalScore + 10 + BonusForSpare(index);
+                    index = index + 2;
+                }
+                // Calculate normal score
+                else
+                {
+                    totalScore = totalScore + CurrentScore(index);
+                    index = index + 2;
+                }
+            }
+
+            return totalScore;
+        }
+
+        public bool IsSpare(int index)
+        {
+            return frame[index] + frame[index + 1] == 10;

```

Comment:
Though index is incremented by 2 after each loop, the method should not rely on how callers handles it. Method must always take care of negative inputs(validations). So here, 
assume input as [ 3 , 4, 6, .........]
what if we call `IsSpare(1)`?
Response from methods will be `true` which is not the right behavior.

This is where OOPS can help.
you can call `currentFrame.IsSpare()` where internally methods will check if the total of 2 pins in currentFrame obj is 10. This is isolation where other frames are not in picture and hence provide good readability and no risk of calculation errors.

Let me know if you need more explanation.